### PR TITLE
Sets up 23.6.0 branch

### PR DESCRIPTION
**Pull request by zoxpx**
Sets up 23.6.0 branch

Signed-off-by: Zoran Rajic <zrajic@purestorage.com>

**What this PR does / why we need it**:

This PR sets up the metadata and build for `px-rel-23.6.0` branch

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,544 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: portworx-operator.v23.5.0
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml:4]: *v23.6.0

File: deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+  namespace: placeholder
+  annotations:
+    capabilities: Auto Pilot
+    categories: "Storage"
+    description: Cloud native storage solution for production workloads
+    containerImage: registry.connect.redhat.com/portworx/openstorage-operator@sha256:744cccea5851eec50efab35c6eb00813388c58c7e5365a3da2463bd53d0ff2d8
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml:10]: docker.io/portworx/px-operator:23.6.0-dev

File: deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+  replaces: portworx-operator.v23.5.0
+  maintainers:
+  - name: Portworx
+    email: support@portworx.com
+  provider:
+    name: Portworx
+  keywords: ["portworx", "persistent storage", "storage", "cloud native", "open source"]
+  relatedImages:
+    - name: portworx-operator
+      image: registry.connect.redhat.com/portworx/openstorage-operator@sha256:744cccea5851eec50efab35c6eb00813388c58c7e5365a3da2463bd53d0ff2d8
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml:65]: docker.io/portworx/px-operator:23.6.0-dev

File: deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+                  - labelSelector:
+                      matchExpressions:
+                      - key: name
+                        operator: In
+                        values:
+                        - portworx-operator
+                    topologyKey: kubernetes.io/hostname
+              containers:
+              - name: portworx-operator
+                image: registry.connect.redhat.com/portworx/openstorage-operator@sha256:744cccea5851eec50efab35c6eb00813388c58c7e5365a3da2463bd53d0ff2d8
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.6.0/portworx-certified.clusterserviceversion.yaml:222]: docker.io/portworx/px-operator:23.6.0-dev

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM after addressing comments

**Comment by zoxpx** [comment]
Thanks Piyush -- merging the PR

**Pull request merged**